File: linear-and-multivariable-algebra/projects/3d-transformation-engine/main.py
import numpy as np

def rotate_z(theta):
    c, s = np.cos(theta), np.sin(theta)
    return np.array([[c, -s, 0], [s, c, 0], [0, 0, 1]])

point = np.array([1, 0, 0])
rotated = np.dot(rotate_z(np.pi/2), point)
print(rotated)

File: linear-and-multivariable-algebra/projects/3d-transformation-engine/main.cpp
#include "../linalg.hpp"

linalg::Matrix rotate_z(double theta) {
    double c = std::cos(theta);
    double s = std::sin(theta);
    return {{c, -s, 0}, {s, c, 0}, {0, 0, 1}};
}

int main() {
    linalg::Vector point = {1, 0, 0};
    linalg::Matrix rot = rotate_z(M_PI / 2.0);
    linalg::Vector rotated = linalg::multiply(rot, point);
    
    linalg::print(rotated, "Rotated Point");
    return 0;
}
